processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -361,8 +365,19 @@ private GroupedTargetReferences groupByTargetReferences( ) {
             Map<String, Set<MappingReference>> singleTargetReferences = new LinkedHashMap<>();
             for ( MappingReference mapping : mappingReferences.getMappingReferences() )  {
                 TargetReference targetReference = mapping.getTargetReference();
+                SourceReference sourceReference = mapping.getSourceReference();
                 String property = first( targetReference.getPropertyEntries() );
                 MappingReference newMapping = mapping.popTargetReference();
+
+                if ( newMapping == null && method.getSourceParameters().size() == 1 && sourceReference != null && sourceReference.getParameter() != null &&
```

Comment:
I think we need to add some examples here in the code. . I'm pretty lost here (so an actual example).

Also we have a lot of checks on whether there are more source-parameters in our bean mapping. WDYT? is it interesting to factor this case out (in the future) in its own `BeanMappingMethod` e.g. `MultipleSourceBeanMappingMethod`?